staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -206,7 +206,7 @@ func (le *LeaderElector) renew(ctx context.Context) {
 	wait.Until(func() {
 		timeoutCtx, timeoutCancel := context.WithTimeout(ctx, le.config.RenewDeadline)
 		defer timeoutCancel()
-		err := wait.PollImmediateUntil(le.config.RetryPeriod, func() (bool, error) {
+		err := wait.PollUntil(le.config.RetryPeriod, func() (bool, error) {
```

Comment:
```go
// PollUntil tries a condition func until it returns true, an error or stopCh is
// closed.
//
// PollUntil always waits interval before the first run of 'condition'.
// 'condition' will always be invoked at least once.
```
If you change this line then you need to increase the timeout in context above by the `le.config.RetryPeriod`. See my original code (only visible via the comment) https://github.com/kubernetes/kubernetes/pull/57932#discussion_r193166191

Reply:
SGTM

Reply:
Or we could revert to PollUntil.